app/models/job_queue.rb

```diff
@@ -23,11 +24,24 @@ def executing?(id)
   end
 
   def queued?(id)
-    @queue.values.detect { |jes| jes.detect { |je| return je if je.id == id } }
+    @queue.each do |i|
```

Comment:
can do `.detect` to avoid the `nil` below

Reply:
how about putting `queue, job_execution` in the queue so these each/detect get simpler ?

Reply:
something like `@queue.each do |queue, job_execution|`

Reply:
Not really understanding these comments. This change would make `@queue` an array of hashes. I'm interpreting this comment as a suggestion to make it a hash with pairs of queue => job_execution which wouldn't work because there could be multiple entries for each queue. I'm probably misunderstanding something.

Reply:
It seems like the easiest way to make this more succinct is to do...

```
@queue.detect  { |i| i[:job_execution].id == id}
```

Then the places that consume `@queued` can be updated to expect the hash with `queue: 'foo', 'job_execution': bar` and extract `job_execution` as needed.

Reply:
Actually it seems like it can just be this...

```
@queue.detect  { |i| return i[:job_execution] if i[:job_execution] == id}
```